### DISCUSSION TITLE
visionOS support

### DISCUSSION THREAD

**Discussion by guygodin** [dotnet/maui#22175] [Ideas]
### Description

Now that the Vision Pro is available publicly to customers, I think adding the ability to target visionOS in .NET MAUI would be very valuable. MAUI already supports targeting iOS, tvOS and watchOS devices.

### Public API Changes

/// <summary>
/// Gets an instance of <see cref="DevicePlatform"/> that represents Apple visionOS.
/// </summary>
public static DevicePlatform visionOS { get; } = new DevicePlatform(nameof(visionOS));

### Intended Use-Case

Having the ability to deploy existing iOS apps to the Vision Pro as Window apps could be a first, easier feature to add. Later updates could add support for volume and eventually immersive space apps as those need to expose a broader range of APIs.

**Comment by iRonJ** [comment]
The spatial computing paradigm in visionOS is still built around windows, but tiled all around the user. The best vision pro productivity apps don’t really need immersive spaces (although this is great for entertainment apps). So as a first iteration, mapping canvases to visionOS windows, and buttons and menus to visionOS toolbars might get most current apps to visionOS without much fuss. Then from there you can iterate as visionOS matures (it’s still missing a lot of useful apis to really do great vision apps).
